File: python/design-patterns/factory/naive/services_naive.py
class OrderService:
    def notify(self, user, msg):
        # ❌ the "which class?" decision is made right here, inline
        notifier = SmsNotifier() if user.prefers_sms else EmailNotifier()
        notifier.send(msg)

class BillingService:
    def remind(self, user, msg):
        # ❌ ...and copy-pasted again here
        notifier = SmsNotifier() if user.prefers_sms else EmailNotifier()
        notifier.send(msg)

# Adding Slack means editing BOTH services — and every other one like them.

File: go/design-patterns/factory/naive/services_naive.go
func (s *OrderService) Notify(user User, msg string) {
    // ❌ the "which class?" decision is made right here, inline
    var notifier Notifier
    if user.PrefersSms {
        notifier = SmsNotifier{}
    } else {
        notifier = EmailNotifier{}
    }
    notifier.Send(msg)
}

func (b *BillingService) Remind(user User, msg string) {
    // ❌ ...and copy-pasted again here
    var notifier Notifier
    if user.PrefersSms {
        notifier = SmsNotifier{}
    } else {
        notifier = EmailNotifier{}
    }
    notifier.Send(msg)
}